# Source: tonistiigi/binfmt patches/buildkit-direct-execve-v9.2/0005-set-script-path-as-argv0-in-shebang-handler.patch
#   https://raw.githubusercontent.com/tonistiigi/binfmt/master/patches/buildkit-direct-execve-v9.2/0005-set-script-path-as-argv0-in-shebang-handler.patch
# Commit: a60e3356e992 ("Bump qemu to v9.2.0", 2025-01-23)
# Applies to: qemu 9.2.4 (see QEMU_VERSION in build-qemu.sh)
# Maintained upstream for the qemu 9.2.x user-mode series (ImageSource API,
# bprm->src.fd); replaces the locally hand-ported 8.2.2 series.
#
From 39f0697cb9d13d8630543209e5bf22148b32b7cb Mon Sep 17 00:00:00 2001
From: Tonis Tiigi <tonistiigi@gmail.com>
Date: Thu, 26 Aug 2021 01:18:32 +0200
Subject: [PATCH 5/7] set script path as argv0 in shebang handler

Signed-off-by: Tonis Tiigi <tonistiigi@gmail.com>
---
 linux-user/linuxload.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/linux-user/linuxload.c b/linux-user/linuxload.c
index 7e66ef0ab9..71f9f46aea 100644
--- a/linux-user/linuxload.c
+++ b/linux-user/linuxload.c
@@ -259,6 +259,8 @@ int load_script(struct linux_binprm *bprm)
      *           (4) user arguments (argv[1:])
      */
 
+    bprm->argv[0] = bprm->filename;
+
     execfd = open(i_name, O_RDONLY);
     if (execfd < 0) {
         perror("script_prepare_binprm: could not open script");
-- 
2.39.3 (Apple Git-146)

